### DISCUSSION TITLE
Interaction between different blocks

### DISCUSSION THREAD

**Discussion by paulordutra** [peridigm/peridigm#350] [General]
Hi, everyone. I'd like to know how Peridigm defines bonds across two different blocks (e. g., two different materials). For example, values for micromodulus and how the damage on bonds is computed by default.

**Comment by djlittl** [comment]
For any pair of nodes (A and B), regardless of whether the are in different material blocks, the Peridigm data structures basically treat the situation as if there are two bonds.  One bond goes from node A to node B and you can think of it as "owned" by node A, and one from node B to node A that is owned by node B.  The pairwise force density for both of these bonds is multiplied by one half.  So, if both A and B are in the same material block, then the net result after the pairwise forces are assembled is exactly what you'd expect.  When A and B are in different materials, the net result is an average of the two pairwise forces.

Bond failure rules work the same way.  Whatever failure rule (if any) that is applied to the material where A is located is applied to the bond from A to B, and vice versa for B.  If the same failure rule is applied to A and B, then the two bonds will fail at the same time (at least, this is true for the critical stretch failure rule).

It would not be too difficult to code a bond failure rule that takes material blocks into account and does something special at the interface.

There isn't much in the literature about how interfaces should be treated.  Nonlocality blends the interface over a region, and this can obviously have some side effects.  For example, averaging the material behavior seems reasonable if the two materials are similar, but makes much less sense if the two materials are very different, say, steel and air.

**Comment by paulordutra** [comment]
OK. Thanks, David. 

In fact, I was trying different boundary conditions (with and without an interface treatment) and found better results without any additional adjust. Since Peridigm doesn't work with rigid bodies, one can either choose the contact model (high computational cost) or propose alternatives damage and material models for the interface. For me, the first method (default) works very well. In the same way, the surface effect decreases.
